tests/test_leftovers.py

```diff
@@ -0,0 +1,294 @@
+    assert res.status_code == 200
+    res = client.get("/index/?size=" + str(data["size"]))
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper size
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["size"] == 100
+
+
+def test_index_list_by_file_name(client, user):
+    # post two records of different name
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["file_name"] = "test_file"
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?file_name=" + data["file_name"])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper name
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["file_name"] == data["file_name"]
+
+
+def test_index_list_by_authz(client, user):
+    # post two records of different authz
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["authz"] = ["test_authz"]
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?authz=" + data["authz"][0])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper authz
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["authz"] == data["authz"]
+
+
+def test_index_list_by_version(client, user):
+    # post two records of different version
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["version"] = "3"
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?version=" + data["version"])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper version
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["version"] == data["version"]
+
+
+def test_negate_filter_file_name(client, user):
+    # post two records of different file name
+    data1 = get_doc()
+    data1["file_name"] = "test_file_name_1"
+    res_1 = client.post("/index/", json=data1, headers=user)
+    assert res_1.status_code == 200
+
+    data2 = get_doc()
+    data2["file_name"] = "test_file_name_2"
+    res_2 = client.post("/index/", json=data2, headers=user)
+    assert res_2.status_code == 200
+
+    negate_param = {"file_name": "test_file_name_2"}
+    res = client.get("/index/?negate_params=" + json.dumps(negate_param))
+    assert res.status_code == 200
+    rec = res.json
+    # assert record returned with proper non-negated file name
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["file_name"] == data1["file_name"]
+
+
+def test_negate_filter_acl(client, user):
```

Comment:
i would argue that you don't need these 4 `test_negate_filter_file_name`, `test_negate_filter_acl`, `test_negate_filter_authz` and `test_negate_filter_version`. there are 3 different cases for this feature: `negate_filter` can be a string, a list or a dict. for dict, there are 2 cases (see details [here](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/Indexd/master/openapis/swagger.yaml#/index/listEntries))

imo you should keep `test_negate_filter_file_name` and `test_negate_filter_authz`, remove the other 2 (unless they are adding coverage -> lmk if i'm wrong) and add a test for either `metadata` or `urls_metadata` that includes both dict use cases

Reply:
these do add to the test coverage, but I can definitely remove them if they don't add any actual value.

Reply:
then let's keep them :)

Reply:
i didn't see the dict use case though

Reply:
note on this: it's in `test_index_list_with_params_negate`